pandas/io/sas/sas7bdat.py

```diff
@@ -54,12 +55,16 @@ def _convert_datetimes(sas_datetimes: pd.Series, unit: str) -> pd.Series:
         if unit == "s":
             s_series = sas_datetimes.apply(
                 lambda sas_float: datetime(1960, 1, 1) + timedelta(seconds=sas_float)
+                if notna(sas_float)
```

Comment:
I'm not a fan of multi-line lambdas. Given the amount of repeated code in the two functions, would it not be better to refactor as a local function that handles both cases?

Reply:
agreed, @wertha if you can make a helper function to do this (module level) and call it

Reply:
done

Reply:
I went back for isnull, since I don't want to double negate the condition or having a nested if, unless you really want to make it clear that is `NaN` I can change it.

Reply:
nvm, i just check there is a `isna`. I updated the code to use it.